### PR TITLE
Add `CodeType.co_lines()` and `CodeType.co_linetable`

### PR DESCRIPTION
**Pull request by AlexWaygood**
PEP 626, introducing these new features, [states](https://www.python.org/dev/peps/pep-0626/#the-new-co-lines-method-of-code-objects):

> ## The new `co_lines()` method of code objects
> The `co_lines()` method will return an iterator which yields tuples of values, each representing the line number of a range of bytecodes. Each tuple will consist of three values:
>
> - `start` -- The offset (inclusive) of the start of the bytecode range
> - `end` -- The offset (exclusive) of the end of the bytecode range
> - `line` -- The line number, or `None` if the bytecodes in the given range do not have a line number.
> The sequence generated will have the following properties:
>
> - The first range in the sequence with have a start of 0
> - The `(start, end)` ranges will be non-decreasing and consecutive. That is, for any pair of tuples the start of the second will equal to the end of the first.
> - No range will be backwards, that is `end >= start` for all triples.
> - The final range in the sequence with have end equal to the size of the bytecode.
> - `line` will either be a positive integer, or None
>
> ## The `co_linetable` attribute
> The `co_linetable` attribute will hold the line number information. The format is opaque, unspecified and may be changed without notice. The attribute is public only to support creation of new code objects.

### CODE REVIEW

File: stdlib/types.pyi [hunk 1]
```diff
@@ -144,6 +144,9 @@ class CodeType:
             co_name: str = ...,
             co_lnotab: bytes = ...,
         ) -> CodeType: ...
+    if sys.version_info >= (3, 10):
```
Comment (Akuli) [inline stdlib/types.pyi:147]: `replace()` above should be updated for Python 3.10 as well.

File: stdlib/types.pyi [hunk 2]
```diff
@@ -123,7 +123,28 @@ class CodeType:
...
+            co_consts: Tuple[Any, ...] = ...,
+            co_names: Tuple[str, ...] = ...,
+            co_varnames: Tuple[str, ...] = ...,
+            co_freevars: Tuple[str, ...] = ...,
+            co_cellvars: Tuple[str, ...] = ...,
+            co_filename: str = ...,
+            co_name: str = ...,
+            co_lnotab: bytes = ...,
+        ) -> CodeType: ...
+    elif sys.version_info >= (3, 10):
```
Comment (Akuli) [inline stdlib/types.pyi:147]: The following structure would be simplest:

```
if sys.version_info >= (3, 10):
    ...
elif sys.version_info >= (3, 8):
    ...
else:
    ...
```

Comment (AlexWaygood) [reply stdlib/types.pyi:147]: *Slaps hand on head*

File: stdlib/types.pyi [hunk 3]
```diff
@@ -123,7 +123,31 @@ class CodeType:
...
+            co_firstlineno: int = ...,
+            co_code: bytes = ...,
+            co_consts: Tuple[Any, ...] = ...,
+            co_names: Tuple[str, ...] = ...,
+            co_varnames: Tuple[str, ...] = ...,
+            co_freevars: Tuple[str, ...] = ...,
+            co_cellvars: Tuple[str, ...] = ...,
+            co_filename: str = ...,
+            co_name: str = ...,
+            co_lnotab: bytes = ...,
```
Comment (Akuli) [inline stdlib/types.pyi:145]: According to stubtest, `lnotab` is gone on 3.10 (see last line of output below). There are several other stubtest errors too, but it would be best to fix them in a separate PR.

```
(env310) akuli@akuli-desktop:~/typeshed$ python3 -m mypy.stubtest --concise --custom-typeshed-dir=. types | grep replace
types.CodeType.replace is inconsistent, runtime argument "co_cellvars" has a default value of type None, which is incompatible with stub argument type builtins.tuple[builtins.str]
types.CodeType.replace is inconsistent, runtime argument "co_code" has a default value of type None, which is incompatible with stub argument type builtins.bytes
types.CodeType.replace is inconsistent, runtime argument "co_consts" has a default value of type None, which is incompatible with stub argument type builtins.tuple[Any]
types.CodeType.replace is inconsistent, runtime argument "co_filename" has a default value of type None, which is incompatible with stub argument type builtins.str
types.CodeType.replace is inconsistent, runtime argument "co_freevars" has a default value of type None, which is incompatible with stub argument type builtins.tuple[builtins.str]
types.CodeType.replace is inconsistent, runtime argument "co_name" has a default value of type None, which is incompatible with stub argument type builtins.str
types.CodeType.replace is inconsistent, runtime argument "co_names" has a default value of type None, which is incompatible with stub argument type builtins.tuple[builtins.str]
types.CodeType.replace is inconsistent, runtime argument "co_varnames" has a default value of type None, which is incompatible with stub argument type builtins.tuple[builtins.str]
types.CodeType.replace is inconsistent, runtime does not have argument "co_lnotab"
```

```suggestion
```

**Review by Akuli — APPROVED** [review]

File: stdlib/types.pyi (hunk 3 above)
Comment (Akuli) [reply stdlib/types.pyi]: Actually, none of these can actually be `None`, even though stubtest thinks so.
